tests/phirl/test_maxent.py

```diff
@@ -0,0 +1,191 @@
+import numpy as np
+import numpy.testing as nptest
+import pytest
+
+import phirl.maxent as me
+
+
+@pytest.mark.parametrize("n_actions", (2, 5, 7))
+def test_n_states(n_actions: int) -> None:
+    n_s = me.get_n_states(n_actions)
+
+    assert isinstance(n_s, int)
+    assert n_s == 2**n_actions
+
+
+@pytest.mark.parametrize("n_actions", (2, 5, 7))
+def test_state_space(n_actions) -> None:
+    space = me.get_state_space(n_actions)
+    size = me.get_n_states(n_actions)
+
+    # Check the size of the space
+    assert len(space) == size
+
+    # Check whether the states are unique
+    assert len(set(space)) == size
+
+    # Check each state separately
+    for state in space:
+        assert isinstance(state, tuple)
+        assert len(state) == n_actions
+
+        # Check if this is a binary vector
+        assert max(state) <= 1
+        assert min(state) >= 0
+
+
+class TestDeterministicTreeMDP:
+    @pytest.mark.parametrize("n_actions", (2, 5))
+    def test_init(self, n_actions: int) -> None:
+        """Tests if the object has proper parameters."""
+        mdp = me.DeterministicTreeMDP(n_actions)
+
+        assert mdp.n_states == me.get_n_states(n_actions)
+        assert mdp.n_actions == n_actions
+
+    def test_new_state_ok(self) -> None:
+        """Tests if the transition function works as expected."""
+        n_actions = 3
+        mdp = me.DeterministicTreeMDP(n_actions)
```

Comment:
The argument `state_space` seems missing: should be changed to `me.DeterministicTreeMDP(n_actions, state_space=me.get_state_space(n_actions))`.

Reply:
In this version of the code, `DeterministicTreeMDP` builds a `state_space` on its own, so that this argument is not needed anymore :)